### ISSUE TITLE
Panic when used with `npm install` modified `yarn.lock` file

### ISSUE THREAD

**Issue by victorb** [robertohuertasm/yarn-lock-parser#3]
Slightly weird issue incoming.

So, discovered the following issue the other day with the npm cli: https://github.com/npm/cli/issues/5126

Basically, running `npm install` inexplicably rewrites any `yarn.lock` file found in the project directly. One thing it does, is wrapping everything (preemptively I think) in double-quotes, as yarn does that _sometimes_ but for what I guess is easy of parser implementation, they do it right away with everything.

So when you try to read a npm-modified yarn.lock file, yarn-lock-parser chokes on the input, as it doesn't expect double-quotes.

I think this will be an issue for non-npm-modified lock files as well, as npm states that yarn sometimes use double-quotes as well, as to escape some names in the lock file. I haven't encountered this in the wild yet myself, so can't say more about it.

To reproduce this, I followed my own steps from the issue linked above, which is: 

1. `cd $(mktemp -d)` Create new temporary directory for a test project
1. `npm init --yes` Create new package.json
1. `npm install --save is-number` Add a dependency
1. `yarn install` Install dependencies via yarn, creating the `yarn.lock` file
1. `cp yarn.lock yarn.lock.original` Save a copy of the original `yarn.lock` file
1. `npm install` Run npm install again which modifies the `yarn.lock` file unexpectedly  
1. `diff yarn.lock yarn.lock.original` show the difference between the npm-modified yarn.lock file with the original one that yarn itself produces

And in addition to that, run the example from the README in this repository in that same directory, leading to the following:

```
$ cat src/main.rs
use std::{error::Error, fs};
use yarn_lock_parser::{parse_str, Entry};

fn main() -> Result<(), Box<dyn Error>> {
    let yarn_lock_text = fs::read_to_string("yarn.lock")?;
    let entries: Vec<Entry> = parse_str(&yarn_lock_text)?;

    println!("{:?}", entries);

    Ok(())
}

$ RUST_BACKTRACE=1 cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/app`
thread 'main' panicked at 'assertion failed: `(left != right)`
  left: `""`,
 right: `""`', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:141:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:182:5
   4: yarn_lock_parser::parse_entry::{{closure}}
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:141:13
   5: core::result::Result<T,E>::map
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/result.rs:756:25
   6: yarn_lock_parser::parse_entry
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:121:5
   7: yarn_lock_parser::entry_final::{{closure}}
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:89:29
   8: core::result::Result<T,E>::map
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/result.rs:756:25
   9: yarn_lock_parser::entry_final
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:88:5
  10: yarn_lock_parser::parse
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:59:28
  11: yarn_lock_parser::parse_str
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/yarn-lock-parser-0.3.0/src/lib.rs:42:5
  12: app::main
             at ./src/main.rs:6:31
  13: core::ops::function::FnOnce::call_once
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

The `yarn.lock` file it's trying to parse is the following:

```
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1

"is-number@^7.0.0":
  "integrity" "sha512-41Cifkg6e8TylSpdtTpeLVMqvSBEVzTttHvERD741+pnZ8ANv0004MRL43QKPDlK9cGvNp6NZWZUBlbGXYxxng=="
  "resolved" "https://registry.npmjs.org/is-number/-/is-number-7.0.0.tgz"
  "version" "7.0.0"
```

**Comment by robertohuertasm** [comment]
I've just tested this and not only it changes the format but also the version. So if you have a v2 yarn.lock file it will be converted to v1.

**Comment by robertohuertasm** [comment]
@victorb fixed in `v0.3.2`. Thanks for reporting this.
